pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -80,61 +82,42 @@ func isInjected(envs map[string]string) bool {
 	return false
 }
 
-func rubyDetector(_ []string, _ map[string]string) Instrumentation {
-	return None
+func getMaps(pid int) ([]byte, error) {
+	mapsPath := kernel.HostProc(strconv.Itoa(pid), "maps")
+	bytes, err := os.ReadFile(mapsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes, nil
 }
```

Comment:
In openssl we have a better implementation that scans the file rather than loading it fully into memory